File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/repository/jpa/BusinessInformationJpaRepository.java
package com.cakk.domain.mysql.repository.jpa;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.cakk.common.enums.VerificationStatus;
import com.cakk.domain.mysql.entity.user.BusinessInformation;
import com.cakk.domain.mysql.entity.user.User;

public interface BusinessInformationJpaRepository extends JpaRepository<BusinessInformation, Long> {

	@Query("select bi from BusinessInformation as bi join fetch bi.cakeShop "
		+ "where bi.cakeShop.id =:cakeShopId")
	Optional<BusinessInformation> findBusinessInformationWithCakeShop(@Param("cakeShopId") Long cakeShopId);

	List<BusinessInformation> findAllByUser(User user);

	Boolean existsBusinessInformationByUserAndCakeShop_Id(User owner, Long cakeShopId);

	@Query("select bi from BusinessInformation as bi join fetch bi.cakeShop where bi.user =:user")
	List<BusinessInformation> findAllWithCakeShopByUser(User user);

	@Query("select bi from BusinessInformation as bi join fetch bi.user join fetch bi.cakeShop"
		+ " where bi.verificationStatus =:verificationStatus")
	List<BusinessInformation> findAllCakeShopBusinessOwnerCandidates(VerificationStatus verificationStatus);

	@Query("select bi from BusinessInformation as bi join fetch bi.user join fetch bi.cakeShop"
		+ " where bi.user.id =:userId")
	Optional<BusinessInformation> findBusinessInformationByUserId(final Long userId);
}

File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/repository/jpa/BusinessInformationJpaRepository.kt
package com.cakk.domain.mysql.repository.jpa

import com.cakk.common.enums.VerificationStatus
import com.cakk.domain.mysql.entity.user.BusinessInformation
import com.cakk.domain.mysql.entity.user.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.*

interface BusinessInformationJpaRepository : JpaRepository<BusinessInformation, Long> {

	@Query("select bi from BusinessInformation as bi join fetch bi.cakeShop "
            + "where bi.cakeShop.id =:cakeShopId")
    fun findBusinessInformationWithCakeShop(@Param("cakeShopId") cakeShopId: Long): BusinessInformation?

	fun findAllByUser(user: User): List<BusinessInformation>

	fun existsBusinessInformationByUserAndCakeShop_Id(owner: User, cakeShopId: Long): Boolean

    @Query("select bi from BusinessInformation as bi join fetch bi.cakeShop where bi.user =:user")
    fun findAllWithCakeShopByUser(user: User): List<BusinessInformation>

    @Query("select bi from BusinessInformation as bi join fetch bi.user join fetch bi.cakeShop"
            + " where bi.verificationStatus =:verificationStatus")
    fun findAllCakeShopBusinessOwnerCandidates(verificationStatus: VerificationStatus): List<BusinessInformation>

    @Query("select bi from BusinessInformation as bi join fetch bi.user join fetch bi.cakeShop"
            + " where bi.user.id =:userId")
    fun findBusinessInformationByUserId(userId: Long): BusinessInformation?
}

